examples/latte/README.md

```diff
@@ -108,12 +121,105 @@ python train.py -c configs/training/sky_video.yaml
+
+The embedding caching process can take a while depending on the size of the video dataset. Some exceptions maybe thrown during the process. If unexpected exceptions are thrown, the program will be stoped and the embedding caching writer's status will be printed on the screen:
+```bash
+Start Video Index: 0. # the start of video index to be processed
+Saving Attempts: 0: save 120 videos, failed 0 videos. # the number of saved video files
+```
+In this case, you can resume the embedding cache from the video indexed at $120$ (index starts from 0). Simply append `--resume_cache_index 120`, and run `python tools/embedding_cache.py`. It will start caching the embedding from the $120^{th}$ video and save the embeddings without overwriting the existing files.
+
+To check more usages, please use `python tools/embedding_cache.py -h`.
+
+</details>
+
+- **Step 2**: Change the dataset configuration file's `data_folder` to the current cache folder path.
+
+After the embeddings have been cached, edit `configs/training/data/sky_numpy_video.yaml`, and change the `data_folder` to the folder where the cached embeddings are stored in.
+
+- **Step 3**: Run the training script.
+
+You can start training on the cached embedding dataset of Sky TimeLapse using:
+```bash
+python train.py -c configs/training/sky_numpy_video.yaml
+```
+
+Note that in `sky_numpy_video.yaml`, we use a large number of frames $128$ and a smaller sample stride $1$, which are different from the settings in `sky_video.yaml` (num_frames=16 and stride=3)· Embedding caching allows us to train Latte to generate more frames with a larger frame rate.
+
+Due to the memory limit, we set the local batch size to $1$ and use a gradient accumulation steps $4$. The number of epochs is $1000$ and the learning rate is $2e^{-5}$. The total number of training steps is about 1000k.
+
+In case of OOM, please set `enable_flash_attention: True` in the `configs/training/sky_numpy_video.yaml`. It can reduce the memory cost and also accelerate the training speed.
+
+### 4.3 Distributed Training
+
+Taking the 4-card distributed training as an example, you can start the distributed training using:
+```bash
+export MS_ASCEND_CHECK_OVERFLOW_MODE="INFNAN_MODE"
+mpirun -n 4 python train.py \
+    -c path/to/configuration/file \
+    --use_parallel True
+```
+where the configuration file can be selected from the `.yaml` files in `configs/training/` folder.
+
+If you have the rank table of Ascend devices, you can take `scripts/run_distributed_sky_numpy_video.sh` as a reference, and start the 4-card distributed training using:
+```bash
+bash scripts/run_distributed_sky_numpy_video.sh path/to/rank/table 0 4
+```
+
+The first number `0` indicates the start index of the training devices, and the second number `4` indicates the total number of distributed processes you want to launch.
+
+
+## 5. Evaluation
+
```

Comment:
it bf16 support? how is the speed/performance again fp16?

Reply:
If setting `patch_embedder` to `linear`, the training can use bf16. I think the speed should be similar to fp16, because they  are both 16 bits.  Bf16 is less likely to lead to NaN compared to fp16.